harwest/lib/codeforces/client.py

```diff
@@ -28,32 +32,18 @@ def get_submissions_page_count(self):
         return int(pages[-1].find('a').text)
 
     def get_submission_code(self, contest_id, submission_id):
-        sub_url = "https://codeforces.com/contest/{contest_id}/submission/{submission_id}".format(
-            contest_id=contest_id,
-            submission_id=submission_id
-        )
-        sub_soup = self.__get_content_soup(sub_url)
-        # For debug purpose
-        # print(sub_url)
-        # open("last_submission_page.html", "w").write(str(sub_soup))
-        submission_code = sub_soup.find('pre', attrs={'id': 'program-source-text'})
-        if submission_code is None:
+        try:
+            return self.cf_parser.get_solution(contest_id, submission_id)
```

Comment:
What would happen if we call `get_solution(...)` on a submission id which is not public or unavailable?

Reply:
ValueError: Incorrect contest_id or submit_id. But as said in #7 we can create auth in session to get the solution. For now Codeforces API doesn't provide any tool to get solutions.

Reply:
Sounds good! Would love to leverage the auth session feature sometime in the future when we decide to support user login on Harwest.